### PR TITLE
Refactors Xenomorph initialize

### PR DESCRIPTION
**Pull request by Birdtalon**
# About the pull request

Refactors /mob/living/carbon/xenomorph/Initialize() removes a few duplicate proc calls, generate_name() called twice, recalculate_actions() called twice etc.

# Explain why it's good for the game

# Testing Photographs and Procedure

<details>
<summary>Screenshots & Videos</summary>

Put screenshots and videos here with an empty line between the screenshots and the `<details>` tags.

</details>

# Changelog

:cl:
code: Refactiors xenomorph initialize & removes some duplicate proc calls
/:cl:

### CODE REVIEW

File: code/modules/mob/living/carbon/xenomorph/Xenomorph.dm [hunk 1]
```diff
@@ -347,157 +347,158 @@
...
 	if(round_statistics && !statistic_exempt)
 		round_statistics.track_new_participant(faction, 1)
-	generate_name()
 
 	// This can happen if a xeno gets made before the game starts
 	if (hive && hive.hive_ui)
 		hive.hive_ui.update_all_xeno_data()
 
-	job = caste.caste_type // Used for tracking the caste playtime
 	Decorate()
```
Comment (fira) [inline code/modules/mob/living/carbon/xenomorph/Xenomorph.dm:502]: why is this here, i'm gonna go insane

File: code/modules/mob/living/carbon/xenomorph/Xenomorph.dm [hunk 2]
```diff
@@ -347,157 +347,158 @@
...
+	add_inherent_verbs()
+	add_abilities()
+	create_reagents(100)
+	regenerate_icons()
+	toggle_xeno_mobhud() //This is a verb, but fuck it, it just werks
+	toggle_xeno_hostilehud()
+	recalculate_everything()
 
-	if(isqueen(src))
-		SStracking.set_leader("hive_[hivenumber]", src)
```
[the remark this answers is not in the document]
Comment (Birdtalon) [reply code/modules/mob/living/carbon/xenomorph/Xenomorph.dm:420]: Ah I probably missed this when picking apart #4941

File: code/modules/mob/living/carbon/xenomorph/Xenomorph.dm (hunk 1 above)
Comment (Birdtalon) [reply code/modules/mob/living/carbon/xenomorph/Xenomorph.dm:502]: Xenomorphs off to their jobs 🕴

Comment (fira) [reply code/modules/mob/living/carbon/xenomorph/Xenomorph.dm:502]: no i meant the Decorate, this is already handled at atom level 😩

Comment (Birdtalon) [reply code/modules/mob/living/carbon/xenomorph/Xenomorph.dm:502]: Oh I see. Yea I actually don't know this is one of those things I didn't touch yet.

**Review by fira — APPROVED** [review]

**Comment by Birdtalon** [comment]
Some bugs to fix.

**Comment by Birdtalon** [comment]
Fixed bug. Accidently passed `hivenumber` instead of `src.hivenumber` to SStracking so we were effectively tracking `null` on evolve.

**Comment by cm13-github** [comment]
This pull request has conflicts, please resolve those before we can evaluate the pull request.

**Comment by cm13-github** [comment]
Conflicts have been resolved. A maintainer will review the pull request shortly.

File: code/modules/mob/living/carbon/xenomorph/Xenomorph.dm [hunk 3]
```diff
@@ -347,157 +347,159 @@
...
+	///Handle transferring things from the old Xeno if we have one in the case of evolve, devolve etc.
+	if(old_xeno)
+		src.nicknumber = old_xeno.nicknumber
+		src.life_kills_total = old_xeno.life_kills_total
+		src.life_damage_taken_total = old_xeno.life_damage_taken_total
+		src.evolution_stored = old_xeno.evolution_stored
 
-	for(var/T in in_hive.hive_inherant_traits)
-		ADD_TRAIT(src, T, TRAIT_SOURCE_HIVE)
+		for(var/datum/language/language in old_xeno.languages)
```
Comment (harryob) [inline code/modules/mob/living/carbon/xenomorph/Xenomorph.dm:374]:
```suggestion
		for(var/datum/language/language as anything in old_xeno.languages)
```
god i hope this list has nothing else in it

File: code/modules/mob/living/carbon/xenomorph/Xenomorph.dm [hunk 4]
```diff
@@ -347,157 +347,159 @@
...
+			add_language(language.name)//Make sure to keep languages (mostly for event Queens that know English)
+
+		//Carry over intents & targeted limb to the new Xeno
+		set_movement_intent(old_xeno.m_intent)
+		a_intent_change(old_xeno.a_intent)
+
+		//We are hiding, let's keep hiding if we can!
+		if(old_xeno.layer == XENO_HIDING_LAYER)
+			for(var/datum/action/xeno_action/onclick/xenohide/hide in actions)
+				if(istype(hide))
```
Comment (harryob) [inline code/modules/mob/living/carbon/xenomorph/Xenomorph.dm:384]: the loop above automatically performs this typecheck, so this is redundant

**Review by harryob — CHANGES_REQUESTED** [review]

File: code/modules/mob/living/carbon/xenomorph/Xenomorph.dm (hunk 4 above)
Comment (Birdtalon) [reply code/modules/mob/living/carbon/xenomorph/Xenomorph.dm]: oh yea missed that one
